cypress/e2e/clue/branch/student_tests/dataflow_tool_spec.js

```diff
@@ -71,6 +71,21 @@ context('Dataflow Tool Tile', function () {
         dataflowToolTile.getNode(nodeType).should("not.exist");
       });
     });
+    describe("Drag to Add Node", () => {
+      const dataTransfer = new DataTransfer;
+      const nodeType = "number";
+      // TODO Why isn't this test working?
+      it.skip('can create node by dragging button onto tile', () => {
+        const draggable = () => cy.get(".program-toolbar [aria-roledescription='draggable'] button").eq(1);
+        dataflowToolTile.getNode(nodeType).should("not.exist");
+        draggable().focus().trigger('dragstart', { dataTransfer });
+        dataflowToolTile.getDataflowTile().trigger('drop', { dataTransfer });
+        draggable().trigger('dragend');
+        dataflowToolTile.getNode(nodeType).should("exist");
+        dataflowToolTile.getDeleteNodeButton(nodeType).click();
+        dataflowToolTile.getNode(nodeType).should("not.exist");
+      });
+    });
```

Comment:
This test is invoking HTML5 drag/drop, not DnDKit drag/drop. @tejal-shah has gotten cypress tests of DnDKit drag/drop working in CODAP3 so you can check there for inspiration.

Reply:
I spent another hour or two this morning trying to get the test to work based on @tejal-shah 's implementation in CODAP, but I couldn't figure it out. I wasn't able to copy it exactly, but I tried several variants that I thought captured the core of how it worked with no success. Ultimately, it seems like the problem is that my `trigger("mousemove")` was not registering the `clientX` and `clientY` I was passing it. Even when I hard coded these values, it looked like the mouse was staying on the element I was trying to drag.

I've left a bare bones version of what I was trying to do in the cypress test. @tejal-shah , if you have some free time, feel free to take a look. Maybe I'm making a mistake that will be obvious to you.

Reply:
The other "trick" is in the context -- you need to enable the `MouseSensor`:
```typScript
interface IProps {
  children: ReactNode
}
export const CodapDndContext = ({ children }: IProps) => {

  const useMouseSensor = useSensor(MouseSensor)
  const sensors = useSensors(
                    // pointer must move three pixels before starting a drag
                    useSensor(PointerSensor, { activationConstraint: { distance: 3 }}),
                    useSensor(KeyboardSensor, { coordinateGetter: customCoordinatesGetter }),
                    // mouse sensor can be enabled for cypress tests, for instance
                    urlParams.mouseSensor !== undefined ? useMouseSensor : null)
  return (
    <DndContext collisionDetection={dndDetectCollision} sensors={sensors} modifiers={[containerSnapToGridModifier]}>
      {children}
    </DndContext>
  )
}
```
Note that in CODAP we hid it behind a URL flag.

Reply:
@tealefristoe , @kswenson , I just now pushed a commit to this branch implementing the `mouseSensor` as a url param in clue (similar to what we did in codap) and was able to successfully get drag and drop to work in your test and so have unskipped that test now in my commit.